src/datahike/norm/norm.cljc

```diff
@@ -0,0 +1,83 @@
+(ns datahike.norm.norm
+  (:require
+    [clojure.java.io :as io]
+    [clojure.string :as string]
+    [taoensso.timbre :as log]
+    [datahike.api :as d]))
+
+(defn attribute-installed? [conn attr]
+  (some? (d/entity @conn [:db/ident attr])))
+
+(defn ensure-norm-attribute! [conn]
+  (if-not (attribute-installed? conn :tx/norm)
+    (:db-after (d/transact conn {:tx-data [{:db/ident       :tx/norm
+                                            :db/valueType   :db.type/keyword
+                                            :db/cardinality :db.cardinality/one}]}))
+    @conn))
+
+(defn norm-installed? [db norm]
+  (->> {:query '[:find (count ?t)
+                 :in $ ?tn
+                 :where
+                 [_ :tx/norm ?tn ?t]]
+        :args  [db norm]}
+       d/q
+       first
+       some?))
```

Comment:
If I am not mistaken, using the dot after count should return a scalar value, if so then you can get rid of the call to `first`

Example:
```
  (d/q '[:find (count ?e)
         :where [?e :weird/stuff]]
       @db)  ;; => []

  (d/q '[:find (count ?e) .
         :where [?e :weird/stuff]]
       @db)  ;; => nil
```

```suggestion
  (->> {:query '[:find (count ?t) .
                 :in $ ?tn
                 :where
                 [_ :tx/norm ?tn ?t]]
        :args  [db norm]}
       d/q
       some?))
```

Reply:
good catch. unfortunately I force-pushed so cannot accept your commit suggestion anymore. But thanks.